### DISCUSSION TITLE
Unable to add fediverse social accounts unless they're currently hosted by Mastodon

### DISCUSSION THREAD

**Discussion by kkostov** [community/community#53537] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Bug

### Body

The Social accounts section of the Profile page allows for users to add their fediverse handles.

The feature seems to be broken for other server types of the fediverse, like Pleroma, Akkoma, Calckey, Pixelfed and others as it doesn't accept the profile link as a social account. As a result, the following issues arise:

* The profile page includes incorrect icon, for mastodon handles you show the mastodon server logo, but for other fediverse servers they appear as a regular link. Ideally, you'd fetch the server icon from the server itself.
* Querying the user's profile via the REST API at `https://api.github.com/users/" + user + "/social_accounts` does not assign the correct `provider` value for social links to fediverse projects. Ideally, one would expect the server type as reported by their rest Api or at least something more generic as "fediverse".
<img width="313" alt="Screenshot 2023-04-23 at 14 12 40" src="https://user-images.githubusercontent.com/4718042/233839292-6ec2aa26-e987-41f8-bcd5-978427c3ac80.png">
<img width="329" alt="Screenshot 2023-04-23 at 14 12 34" src="https://user-images.githubusercontent.com/4718042/233839293-df02797f-1b8e-4767-8099-cd0323c81044.png">

**Comment by davidgarywood** [comment]
Yup this is ignoring how broad the Fediverse is. I’ve just registered a CalcKey account as well as Mastodon and can’t link that here properly.

**Reply by jsailo** [reply]
I too have moved to a Rebased + Soapbox site. Had decided on Akkoma but their Admin picked an SS of my other query in their forums and pasted in another so I'm done with them. And oh gosh! Their 'Yo, I was on Easter holiday' type of emails that I never signed up for.

**Comment by setzor** [comment]
Calckey (and basically the rest of the fediverse) getting side-lined here ☹️

**Comment by hellquist** [comment]
I noticed I could save links with querystrings though. This is what the link to my Akkoma account now looks like: https://thias.hellqui.st/m?Info=Fediverse_Account_Compatible_With_Mastodon

**Comment by crse** [comment]
Still waiting support for Misskey, Pleroma, and Lemmy.

Or alternatively, change Mastodon logo altogether to be a universal fediverse logo.
